File: agent/tools/jira_get_transitions.py
import asyncio
from typing import Any

from ..utils.jira import get_transitions


def jira_get_transitions(issue_key: str) -> dict[str, Any]:
    """Get available workflow transitions for a Jira issue.

    Use this to see what status changes are available before calling jira_transition_issue.

    Args:
        issue_key: The issue key (e.g., "PROJ-123")

    Returns:
        Dictionary containing:
        - transitions: List of available transitions with:
            - id: Transition ID (needed for jira_transition_issue)
            - name: Display name of the transition
            - to: Target status information
        - error: Error message if the request failed
    """
    return asyncio.run(get_transitions(issue_key))

File: src/tools/jiraGetTransitions.ts
/**
 * jira_get_transitions tool
 * Mirrors agent/tools/jira_get_transitions.py
 */

import { tool } from "@langchain/core/tools";
import { z } from "zod";
import { getTransitions } from "../utils/jira.js";

export const jiraGetTransitions = tool(
  async ({ issueKey }) => {
    try {
      const result = await getTransitions(issueKey);
      return JSON.stringify({ success: true, transitions: result });
    } catch (err) {
      return JSON.stringify({ error: String(err), status: "error" });
    }
  },
  {
    name: "jira_get_transitions",
    description:
      "Get the available status transitions for a Jira issue. " +
      "Use jira_transition_issue with the returned transition ID to change the status.",
    schema: z.object({
      issueKey: z.string().describe("The Jira issue key, e.g. 'PROJ-123'."),
    }),
  },
);
